### PR TITLE
chore(storage-browser): Add control hooks for consistency

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

This PR lightly refactors Pagination and Search to follow other controls' patterns

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/controls/__tests__/PaginationControl.spec.tsx [hunk 1]
```diff
@@ -1,60 +1,34 @@
...
-    const nav = screen.getByRole('navigation');
-    const list = screen.getByRole('list');
-    const listItems = await screen.findAllByRole('listitem');
-    const nextButton = screen.getByRole('button', { name: 'Go to next page' });
-    const prevButton = screen.getByRole('button', {
-      name: 'Go to previous page',
-    });
-    const nextIcon = nextButton.querySelector('svg');
-    const prevIcon = nextButton.querySelector('svg');
+    const searchSubfoldersToggle = screen.getByTestId('pagination');
```
Comment (jordanvn) [inline packages/react-storage/src/components/StorageBrowser/controls/__tests__/PaginationControl.spec.tsx:30]: Holdover variable name from copied file?

File: packages/react-storage/src/components/StorageBrowser/controls/__tests__/SearchFieldControl.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  });
+
+  afterEach(() => {
+    mockUseSearchField.mockClear();
+  });
+
+  it('renders', () => {
+    render(<SearchFieldControl />);
+
+    const searchSubfoldersToggle = screen.getByTestId('search-field');
```
Comment (jordanvn) [inline packages/react-storage/src/components/StorageBrowser/controls/__tests__/SearchFieldControl.spec.tsx:30]: Should this variable name be changed?

**Review by jordanvn — COMMENTED** [review]
Looks good to me, just some variables that seem misnamed

File: packages/react-storage/src/components/StorageBrowser/controls/__tests__/PaginationControl.spec.tsx (hunk 1 above)
Comment (cshfang) [reply packages/react-storage/src/components/StorageBrowser/controls/__tests__/PaginationControl.spec.tsx:30]: Totes. Ty

**Pull request closed without merging**